src/main/java/net/sacredlabyrinth/phaed/simpleclans/ClanPlayer.java

```diff
@@ -581,19 +575,12 @@ public String getPastClansString(@NotNull String sep) {
      */
     @NotNull
     public String getPastClansString(String sep, @Nullable Player viewer) {
-        StringBuilder out = new StringBuilder();
-
-        for (String pastClan : getPastClans()) {
-            out.append(pastClan).append(sep);
-        }
-
-        out = new StringBuilder(Helper.stripTrailing(out.toString(), sep));
-
-        if (out.toString().trim().isEmpty()) {
+        if (getPastClans().isEmpty()) {
             return lang("none", viewer);
         }
 
-        return out.toString();
+        SettingsManager settings = SimpleClans.getInstance().getSettingsManager();
+        return getPastClans().stream().limit(settings.getInt(PAST_CLANS_LIMIT)).collect(Collectors.joining(sep));
```

Comment:
This will _probably_ show the first _n_ clans. It should be the last clans. (reverse order)